src/sentry/search/snuba/executors.py

```diff
@@ -404,15 +390,15 @@ def query(
             if not snuba_groups:
                 break
 
-            if candidate_ids:
+            if group_ids:
                 # pre-filtered candidates were passed down to Snuba, so we're
                 # finished with filtering and these are the only results. Note
                 # that because we set the chunk size to at least the size of
-                # the candidate_ids, we know we got all of them (ie there are
+                # the group_ids, we know we got all of them (ie there are
                 # no more chunks after the first)
                 result_groups = snuba_groups
-                if count_hits and hits is None:
-                    hits = len(snuba_groups)
+                # if count_hits and hits is None:
+                # hits = len(snuba_groups)
```

Comment:
Why is this code path not there anymore ?

Reply:
It's been brought into the calculate_hits function. `hits` is only assigned in one place now.

Reply:
Scratch that. This was reintroduced and we now may assign hits here, if we didn't calculate it before the loop.

This is so we have a chance at returning an empty result set before running the full snuba query.